src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -64,6 +75,7 @@ public void updateById(Long id, CreateBookRequestDto requestDto) {
         book.setPrice(requestDto.getPrice());
         book.setDescription(requestDto.getDescription());
         book.setCoverImage(requestDto.getCoverImage());
+        book.setCategories(book.getCategories());
```

Comment:
```suggestion
        bookMapper.updateBookFromDto(bookDto, book);
        book.setCategories(book.getCategories());
```

use mapper to set up book